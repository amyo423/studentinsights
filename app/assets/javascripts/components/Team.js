import React from 'react';
import PropTypes from 'prop-types';

const TEAM_ICON_MAP = {
  'Competitive Cheerleading Varsity': '📣',
  'Cross Country - Boys Varsity': '👟',
  'Cross Country - Girls Varsity': '👟',
  'Football Varsity': '🏈',
  'Golf Varsity': '⛳',
  'Soccer - Boys Freshman': '⚽',
  'Soccer - Boys JV': '⚽',
  'Soccer - Boys Varsity': '⚽',
  'Soccer - Girls JV': '⚽',
  'Soccer - Girls Varsity': '⚽',
  'Volleyball - Girls Freshman': '🏐',
  'Volleyball - Girls JV': '🏐',
  'Volleyball - Girls Varsity': '🏐'
};


export default function Team({team, style}) {
  return (
    <span title={`${team.activity_text} with ${team.coach_text}`}>
      <TeamIcon teamKey={team.activity_text} style={{paddingRight: 5}} />
      {parseTeam(team.activity_text)}
    </span>
  );
}
Team.propTypes = {
  team: PropTypes.shape({
    activity_text: PropTypes.string.isRequired,
    coach_text: PropTypes.string.isRequired
  }).isRequired,
  style: PropTypes.object
};


export function TeamIcon({teamKey, style}) {
  const emoji = TEAM_ICON_MAP[teamKey] || '🏅';
  return <span style={style}>{emoji}</span>;
}
TeamIcon.propTypes = {
  teamKey: PropTypes.string.isRequired,
  style: PropTypes.object
};


export function parseTeam(activityText) {
  return activityText
    .replace(' - ', ' ')
    .replace('Boys', '')
    .replace('Girls', '')
    .replace('Varsity', '')
    .replace('JV', '')
    .replace('Freshman', '')
    .trim();
}