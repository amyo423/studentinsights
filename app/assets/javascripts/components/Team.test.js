import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Team, {parseTeam} from './Team';

const teams = [
  { coach_text: 'Fatima Teacher', activity_text: 'Competitive Cheerleading Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Cross Country - Boys Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Cross Country - Girls Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Football Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Golf Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Soccer - Boys Freshman' },
  { coach_text: 'Fatima Teacher', activity_text: 'Soccer - Boys JV' },
  { coach_text: 'Fatima Teacher', activity_text: 'Soccer - Boys Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Soccer - Girls JV' },
  { coach_text: 'Fatima Teacher', activity_text: 'Soccer - Girls Varsity' },
  { coach_text: 'Fatima Teacher', activity_text: 'Volleyball - Girls Freshman' },
  { coach_text: 'Fatima Teacher', activity_text: 'Volleyball - Girls JV' },
  { coach_text: 'Fatima Teacher', activity_text: 'Volleyball - Girls Varsity' }
];

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<div>{teams.map(team => <Team key={team.activity_text} team={team} />)}</div>, el);
});

it('snapshots view', () => {
  const tree = renderer
    .create(<div>{teams.map(team => <Team key={team.activity_text} team={team} />)}</div>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('#parseTeam', () => {
  expect(parseTeam('Cross Country - Boys Varsity')).toEqual('Varsity Cross Country');
});
