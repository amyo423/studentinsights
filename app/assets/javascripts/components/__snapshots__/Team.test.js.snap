// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div>
  <span
    title="Competitive Cheerleading Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Competitive Cheerleading Varsity with Fatima Teacher"
    >
      📣
    </span>
    Varsity Competitive Cheerleading
  </span>
  <span
    title="Cross Country - Boys Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Cross Country - Boys Varsity with Fatima Teacher"
    >
      👟
    </span>
    Varsity Cross Country
  </span>
  <span
    title="Cross Country - Girls Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Cross Country - Girls Varsity with Fatima Teacher"
    >
      👟
    </span>
    Varsity Cross Country
  </span>
  <span
    title="Football Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Football Varsity with Fatima Teacher"
    >
      🏈
    </span>
    Varsity Football
  </span>
  <span
    title="Golf Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Golf Varsity with Fatima Teacher"
    >
      ⛳
    </span>
    Varsity Golf
  </span>
  <span
    title="Soccer - Boys Freshman with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Soccer - Boys Freshman with Fatima Teacher"
    >
      ⚽
    </span>
    Freshman Soccer
  </span>
  <span
    title="Soccer - Boys JV with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Soccer - Boys JV with Fatima Teacher"
    >
      ⚽
    </span>
    JV Soccer
  </span>
  <span
    title="Soccer - Boys Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Soccer - Boys Varsity with Fatima Teacher"
    >
      ⚽
    </span>
    Varsity Soccer
  </span>
  <span
    title="Soccer - Girls JV with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Soccer - Girls JV with Fatima Teacher"
    >
      ⚽
    </span>
    JV Soccer
  </span>
  <span
    title="Soccer - Girls Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Soccer - Girls Varsity with Fatima Teacher"
    >
      ⚽
    </span>
    Varsity Soccer
  </span>
  <span
    title="Volleyball - Girls Freshman with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Volleyball - Girls Freshman with Fatima Teacher"
    >
      🏐
    </span>
    Freshman Volleyball
  </span>
  <span
    title="Volleyball - Girls JV with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Volleyball - Girls JV with Fatima Teacher"
    >
      🏐
    </span>
    JV Volleyball
  </span>
  <span
    title="Volleyball - Girls Varsity with Fatima Teacher"
  >
    <span
      style={
        Object {
          "cursor": "default",
          "paddingRight": 5,
        }
      }
      title="Volleyball - Girls Varsity with Fatima Teacher"
    >
      🏐
    </span>
    Varsity Volleyball
  </span>
</div>
`;
